### PR TITLE
Fix: Exception when Portnumber was too big

### PR DESCRIPTION
**Pull request by pascaldornfeld**
The app crashed when the port was bigger than Int.MAX_VALUE

### CODE REVIEW

**Comment by Nicba1010** [comment]
It would be better if you limit the port number to the actual port limit which is 65535 and alert the user if it is bad.

**Comment by pascaldornfeld** [comment]
There you go

File: app/src/main/java/com/aconno/sensorics/ui/settings/virtualscanningsources/MqttVirtualScanningSourceActivity.kt [hunk 1]
```diff
@@ -240,13 +243,26 @@ class MqttVirtualScanningSourceActivity : BaseActivity() {
         val id = if (mqttVirtualScanningSourceModel == null) 0 else mqttVirtualScanningSourceModel!!.id
         val enabled = mqttVirtualScanningSourceModel?.enabled != false
 
+        val portAsInteger = try {
+            port.toInt().also {
+                if(it < 0 || it > MAX_PORT_NUMBER) throw NumberFormatException("Port invalid")
```
Comment (borna-mdn) [inline app/src/main/java/com/aconno/sensorics/ui/settings/virtualscanningsources/MqttVirtualScanningSourceActivity.kt:248]: There is no need for try/catch here, as you could use if statement. Also NumberFormatException is not a correct exception to be thrown in this situation - everything is fine with the number format, but the number is out of range.

You should put the validation in port's EditTextView like Roberto suggested. Then you don't need this part of the code.

File: app/src/main/java/com/aconno/sensorics/ui/settings/virtualscanningsources/MqttVirtualScanningSourceActivity.kt [hunk 2]
```diff
@@ -23,8 +23,11 @@ import kotlinx.android.synthetic.main.mqtt_virtual_scanning_source_form.*
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.GlobalScope
 import kotlinx.coroutines.launch
+import timber.log.Timber
 import javax.inject.Inject
 
+private const val MAX_PORT_NUMBER = 0xFFFF
```
Comment (borna-mdn) [inline app/src/main/java/com/aconno/sensorics/ui/settings/virtualscanningsources/MqttVirtualScanningSourceActivity.kt:29]: You should not put a constant here.

File: app/src/main/java/com/aconno/sensorics/ui/settings/virtualscanningsources/MqttVirtualScanningSourceActivity.kt (hunk 1 above)
Comment (pascaldornfeld) [reply app/src/main/java/com/aconno/sensorics/ui/settings/virtualscanningsources/MqttVirtualScanningSourceActivity.kt]: port is a String. It was possible to make it very long so port.toInt() throws a NumberFormatException.
Now I restricted EditTextView to 5 digits and removed the try-catch.

File: app/src/main/java/com/aconno/sensorics/ui/settings/virtualscanningsources/MqttVirtualScanningSourceActivity.kt (hunk 2 above)
Comment (pascaldornfeld) [reply app/src/main/java/com/aconno/sensorics/ui/settings/virtualscanningsources/MqttVirtualScanningSourceActivity.kt]: Moved it.

File: app/src/main/java/com/aconno/sensorics/ui/settings/virtualscanningsources/MqttVirtualScanningSourceActivity.kt (hunk 1 above)
Comment (borna-mdn) [reply app/src/main/java/com/aconno/sensorics/ui/settings/virtualscanningsources/MqttVirtualScanningSourceActivity.kt]: You should not show a toast, but you should put a validation on EditText.

https://medium.com/@bendaniel10/mini-project-android-edittext-validation-9bd8ab04576a

File: app/src/main/java/com/aconno/sensorics/ui/settings/virtualscanningsources/MqttVirtualScanningSourceActivity.kt [hunk 3]
```diff
@@ -240,13 +242,23 @@ class MqttVirtualScanningSourceActivity : BaseActivity() {
         val id = if (mqttVirtualScanningSourceModel == null) 0 else mqttVirtualScanningSourceModel!!.id
         val enabled = mqttVirtualScanningSourceModel?.enabled != false
 
+        val portAsInteger = port.toInt()
+        if (portAsInteger < 0 || portAsInteger > maxPortNumber) {
+            Toast.makeText(
```
Comment (borna-mdn) [inline app/src/main/java/com/aconno/sensorics/ui/settings/virtualscanningsources/MqttVirtualScanningSourceActivity.kt:247]: Toast is not a good solution.

Comment (pascaldornfeld) [reply app/src/main/java/com/aconno/sensorics/ui/settings/virtualscanningsources/MqttVirtualScanningSourceActivity.kt]: Good idea! Done.

Comment (pascaldornfeld) [reply app/src/main/java/com/aconno/sensorics/ui/settings/virtualscanningsources/MqttVirtualScanningSourceActivity.kt]: Changed it to edittext.error

File: app/src/main/java/com/aconno/sensorics/ui/settings/virtualscanningsources/MqttVirtualScanningSourceActivity.kt (hunk 1 above)
Comment (pascaldornfeld) [reply app/src/main/java/com/aconno/sensorics/ui/settings/virtualscanningsources/MqttVirtualScanningSourceActivity.kt]: Good idea! Changed it to edittext.error

**Pull request merged**